codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordEncoder.java

```diff
@@ -49,29 +53,87 @@ public void encodeRecord(DnsRecord record, ByteBuf out) throws Exception {
             encodeQuestion((DnsQuestion) record, out);
         } else if (record instanceof DnsPtrRecord) {
             encodePtrRecord((DnsPtrRecord) record, out);
+        } else if (record instanceof DnsOptEcsRecord) {
+            encodeOptEcsRecord((DnsOptEcsRecord) record, out);
+        } else if (record instanceof DnsOptPseudoRecord) {
+            encodeOptPseudoRecord((DnsOptPseudoRecord) record, out);
         } else if (record instanceof DnsRawRecord) {
             encodeRawRecord((DnsRawRecord) record, out);
         } else {
             throw new UnsupportedMessageTypeException(StringUtil.simpleClassName(record));
         }
     }
 
-    private void encodePtrRecord(DnsPtrRecord record, ByteBuf out) throws Exception {
+    private void encodeRecord0(DnsRecord record, ByteBuf out) throws Exception {
         encodeName(record.name(), out);
-
         out.writeShort(record.type().intValue());
         out.writeShort(record.dnsClass());
         out.writeInt((int) record.timeToLive());
+    }
 
+    private void encodePtrRecord(DnsPtrRecord record, ByteBuf out) throws Exception {
+        encodeRecord0(record, out);
         encodeName(record.hostname(), out);
     }
 
-    private void encodeRawRecord(DnsRawRecord record, ByteBuf out) throws Exception {
-        encodeName(record.name(), out);
+    private void encodeOptPseudoRecord(DnsOptPseudoRecord record, ByteBuf out) throws Exception {
+        encodeRecord0(record, out);
+        out.writeShort(0);
+    }
 
-        out.writeShort(record.type().intValue());
-        out.writeShort(record.dnsClass());
-        out.writeInt((int) record.timeToLive());
+    private void encodeOptEcsRecord(DnsOptEcsRecord record, ByteBuf out) throws Exception {
+        encodeRecord0(record, out);
+
+        int sourcePrefixLength = record.sourcePrefixLength();
+        int scopePrefixLength = record.scopePrefixLength();
+        int lowOrderBitsToPreserve = sourcePrefixLength & PREFIX_MASK;
+
+        byte[] bytes = record.address();
+        int addressBits = bytes.length << 3;
+        if (addressBits < sourcePrefixLength || sourcePrefixLength < 0) {
+            throw new IllegalArgumentException(sourcePrefixLength + ": "
+                    + sourcePrefixLength + " (expected: 0 >= " + addressBits + ')');
```

Comment:
Didn't we wrap after an operator? i.e

    throw new ...(... +
            sourcePrefixLength + ...);